gptel-transient.el

```diff
@@ -869,6 +913,302 @@ When LOCAL is non-nil, set the system message only in the current buffer."
 
+;; ** Suffix for displaying and removing context
+
+(defun gptel--context-edge-point (edge-type direction &optional inclusive)
+  "Find context edge point of EDGE-TYPE from current point.
+EDGE-TYPE is either :start or :end.
+DIRECTION is either :next or :previous.
+If INCLUSIVE is non-nil, return the current point if it is on an edge."
+  (let* ((point nil)
+         (get-edge-point #'(lambda (direction)
+                             (if (and inclusive
+                                      (when (get-text-property (point) 'gptel-context-overlay)
+                                        (if (eq edge-type :start)
+                                            (when (and (/= (point) (point-min))
+                                                       (not (get-text-property
+                                                             (1- (point))
+                                                             'gptel-context-overlay)))
+                                              t)
+                                          (when (and (/= (point) (point-max))
+                                                     (not (get-text-property
+                                                           (1+ (point))
+                                                           'gptel-context-overlay)))
+                                            t))))
+                                 (point)
+                               (if (eq direction :next)
+                                   (setq point (next-single-property-change
+                                                (point)
+                                                'gptel-context-overlay nil nil))
+                                 (setq point (previous-single-property-change
+                                                (point)
+                                                'gptel-context-overlay nil nil)))))))
+    (save-excursion
+      (if (eq edge-type :end)
+          (progn
+            (funcall get-edge-point direction)
+            (when point
+              (goto-char point)
+              (if (get-text-property (1+ point) 'gptel-context-overlay)
+                  ;; This is actually a starting edge, not an ending edge.
+                  (funcall get-edge-point direction)
+                point)))
+        (funcall get-edge-point direction)
+        (when point
+          (goto-char point)
+          (if (get-text-property (1- point) 'gptel-context-overlay)
+              ;; This is actually an ending edge, not a starting edge.
+              (funcall get-edge-point direction)))))
+    ;; Handle some edge cases (pun unintended).
+    (unless point
+      (if (eq edge-type :end)
+          (when (get-text-property (max (point-min) (1- (point))) 'gptel-context-overlay)
+            (setq point (point)))
+        (when (get-text-property (min (point-max) (1+ (point))) 'gptel-context-overlay)
+            (setq point (point)))))
+    point))
+
+(let* ((highlight-start nil)
```

Comment:
This let block should be inside the suffix function below, unless there's some reason you want these local variables to persist after/between calls to `gptel--suffix-context-buffer`.

Reply:
Right. Not sure why I did it like this...